packages/angular_devkit/build_angular/src/builders/browser-esbuild/index.ts

```diff
@@ -655,6 +655,7 @@ export async function* buildEsbuildBrowserInternal(
       // Ignore all node modules directories to avoid excessive file watchers.
       // Package changes are handled below by watching manifest and lock files.
       '**/node_modules/**',
+      '**/.git/**',
```

Comment:
Maybe we should ignore all dotted directories?

Reply:
Can't think of any that would cause the same problem as the .git folder.

Reply:
@romeomihalovics, one cases could be the cache directory which by default is named `.angular` for example.

Reply:
I've replaced "git" with "*" then, this way every folder starting with a dot should be ignored